libs/content/workspace/src/data.rs

```diff
@@ -0,0 +1,191 @@
+use lb_rs::blocking::Lb;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+
+pub type Graph = Vec<LinkNode>;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct LinkNode {
+    pub id: usize,
+    pub title: String,
+    pub links: Vec<usize>,
+    pub color: [f32; 3],
+    pub cluster_id: Option<usize>,
+    pub internal: bool,
+
+    pub x: f32,
+    pub y: f32,
+    pub vx: f32,
+    pub vy: f32,
+    pub fx: Option<f32>,
+    pub fy: Option<f32>,
+}
+
+#[derive(Clone, Debug)]
+pub struct NameId {
+    pub id: usize,
+    pub name: String,
+    pub links: Vec<usize>,
+    pub internal: bool,
+}
+
+impl NameId {
+    fn new(id: usize, name: String, links: Vec<usize>) -> Self {
+        NameId { id, name, links, internal: true }
+    }
+}
+
+impl LinkNode {
+    fn new(id: usize, title: String, links_given: Vec<usize>) -> Self {
+        LinkNode {
+            id,
+            title,
+            links: links_given.clone(),
+            color: [0.0, 0.0, 0.0],
+            cluster_id: None,
+            internal: true,
+
+            x: 0.0, // Set to an initial value, e.g., random or based on index
```

Comment:
the comment seems confusing. you're setting it to `0.0` but suggesting that the value is random.